code/game/objects/items/devices/hologram_projector.dm

```diff
@@ -0,0 +1,188 @@
+	// Role set to No or Never
+	recruiter.player_not_volunteering = new /callback(src, .proc/recruiter_not_recruiting)
+
+	recruiter.recruited = new /callback(src, .proc/recruiter_recruited)
+	recruiter.request_player()
+
+/obj/item/device/hologram_projector/proc/recruiter_recruiting(mob/dead/observer/player, controls)
+	to_chat(player, "<span class='recruit'>\ You have been added to the list of potential ghosts. ([controls])</span>")
+
+/obj/item/device/hologram_projector/proc/recruiter_not_recruiting(mob/dead/observer/player, controls)
+	to_chat(player, "<span class='recruit'>\ A hologram is being requested. ([controls])</span>")
+
+/obj/item/device/hologram_projector/proc/recruiter_recruited(mob/dead/observer/player)
+	if(!player)
+		to_chat(usr, "Hologram generation failed!")
+		polling_ghosts = FALSE
+		qdel(recruiter)
+		recruiter = null
+		return
+	polling_ghosts = FALSE
+
+	var/turf/T = get_turf(src)
+	player.forceMove(T)
+	var/mob/living/simple_animal/hologram/advanced/projector/H = player.transmogrify(/mob/living/simple_animal/hologram/advanced/projector, TRUE)
+	holoperson = H
+	holoperson.set_light(1)
+	holoperson.projector = src
+	holoperson.proj_turf = T
+
+	icon_state = "shield1"
+	ray = new(T)
+	qdel(recruiter)
+
+/obj/item/device/hologram_projector/examine(mob/user, var/size = "")
+	recruit_holoperson()
+
+/mob/living/simple_animal/hologram/advanced/projector/Life()
+	regular_hud_updates()
+
+	if(!projector)
+		return
+	if((mind && !client))
+		projector.clear_holo()
+
+	var/turf/T = get_turf(src)
+	var/turf/dest = get_turf(projector)
+
+	if(T && T.obscured)
+		projector.clear_holo()
+	if((projector.holo_mode == 0 && (get_dist(dest, T) <= projector.holo_range)))
+		return 1
+	else if (projector.holo_mode == 1)
+		var/area/area = get_area(dest)
+		var/area/holoperson_area = get_area(T)
+		if(holoperson_area == area)
+			return 1
+
+	projector.clear_holo() //If not, we want to get rid of the hologram.
+	
+/mob/living/simple_animal/hologram/advanced/projector/Move()
+	..()
+	if(!projector)
+		return
+	var/turf/T = get_turf(src)
+	var/turf/dest = get_turf(projector)
+	if(proj_turf && proj_turf != dest)
+		qdel(projector.ray)
+		projector.ray = new(dest)
+		proj_turf = dest
+	else
+		var/disty = y - projector.ray.y
+		var/distx = x - projector.ray.x
+		var/newangle
+		if(!disty)
+			if(distx >= 0)
+				newangle = 90
+			else
+				newangle = 270
+		else
+			newangle = arctan(distx/disty)
+			if(disty < 0)
+				newangle += 180
+			else if(distx < 0)
+				newangle += 360
+		var/matrix/M = matrix()
+		if (get_dist(T,dest) <= 1)
+			animate(projector.ray, transform = turn(M.Scale(1,sqrt(distx*distx+disty*disty)),newangle),time = 1)
+		else
+			projector.ray.transform = turn(M.Scale(1,sqrt(distx*distx+disty*disty)),newangle)
+
+//pAI it uses the pAI framework in objs.dm. Check that code for further information
+/obj/item/device/hologram_projector/install_pai(obj/item/device/paicard/P)
```

Comment:
this is not a required change but merely a recommendation: I think it would be more interesting if the pAI could choose turn off the projector so that they can revert to just riding along inside the card (inside the projector itself) - mostly relevant for being stealthy with their master